services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.controller.ts

```diff
@@ -1,116 +1,324 @@
+  public async validateAndCreateProperties(
+    @Body(
+      new ParseArrayPipe({
+        items: CreateProgramFinancialServiceProviderConfigurationPropertyDto,
+      }),
+    )
+    properties: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+    @Param('programId', ParseIntPipe)
+    programId: number,
+    @Param('name')
+    name: string,
+  ): Promise<
+    ProgramFinancialServiceProviderConfigurationResponsePropertyDto[]
+  > {
+    // Should this work with an array of properties or just a single property?
+    return await this.programFinancialServiceProviderConfigurationsService.validateAndCreateProperties(
+      {
+        programId,
+        programFspConfigurationName: name,
+        properties,
+      },
+    );
+  }
+
+  @AuthenticatedUser({ isAdmin: true })
+  @ApiOperation({
+    summary: `Update a single property for a Program Financial Service Provider Configuration.. See ${EXTERNAL_API.baseApiUrl}/financial-service-providers for allowed properties per financial service provider.`,
+  })
+  @ApiParam({ name: 'programId', required: true, type: 'integer' })
+  @ApiParam({
+    name: 'name',
+    required: true,
+    type: 'string',
+  })
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description:
+      'The Financial Service Provider Configuration property has been successfully updated.',
+  })
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Bad request. Body or params are malformed',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description:
+      'Program does not exist or Financial Service Provider Configuration or propery does not exist',
+  })
+  @Patch(
+    ':programId/financial-service-provider-configurations/:name/properties/:propertyName',
+  )
+  public async updateProperty(
+    @Body()
+    property: UpdateProgramFinancialServiceProviderConfigurationPropertyDto,
+    @Param('programId', ParseIntPipe)
+    programId: number,
+    @Param('name')
+    name: string,
+    @Param('propertyName')
+    propertyName: WrapperType<FinancialServiceProviderConfigurationProperties>,
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponsePropertyDto> {
+    return await this.programFinancialServiceProviderConfigurationsService.updateProperty(
+      {
+        programId,
+        programFspConfigurationName: name,
+        propertyName,
+        property,
+      },
+    );
+  }
+
+  @AuthenticatedUser({ isAdmin: true })
+  @ApiOperation({
+    summary: `Delete a single Program Financial Service Provider Configuration property. See ${EXTERNAL_API.baseApiUrl}/financial-service-providers for required properties per financial service provider.`,
+  })
+  @ApiParam({ name: 'programId', required: true, type: 'integer' })
+  @ApiParam({
+    name: 'name',
+    required: true,
+    type: 'string',
+  })
+  @ApiParam({
+    name: 'propertyName',
+    required: true,
+    type: 'string',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
```

Comment:
shouldn't this be 204 given that nothing is returned (see e.g. https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete). Also check other DELETE endpoints.

Reply:
the SO post says "A successful response SHOULD be 200 (OK) if the response includes an entity describing the status, 202 (Accepted) if the action has not yet been enacted, or 204 (No Content) if the action has been enacted but the response does not include an entity." So it should be 204 I think.

Reply:
Ah you're right :)